airbyte-integrations/connectors/source-xero/source_xero/components.py

```diff
@@ -0,0 +1,196 @@
+#
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+#
+
+import base64
+import decimal
+import re
+from dataclasses import InitVar, dataclass
+from datetime import date, datetime, time, timedelta, timezone
+from http import HTTPStatus
+from typing import Any, Dict, List, Mapping, MutableMapping, Optional, Union
+
+import dpath.util
+import pendulum
+import requests
+from airbyte_cdk.sources.declarative.auth.declarative_authenticator import NoAuth
+from airbyte_cdk.sources.declarative.decoders.decoder import Decoder
+from airbyte_cdk.sources.declarative.decoders.json_decoder import JsonDecoder
+from airbyte_cdk.sources.declarative.extractors.record_extractor import RecordExtractor
+from airbyte_cdk.sources.declarative.incremental import DatetimeBasedCursor
+from airbyte_cdk.sources.declarative.interpolation import InterpolatedString
+from airbyte_cdk.sources.declarative.types import Config, StreamSlice, StreamState
+from requests import HTTPError
+
+
+@dataclass
+class CustomExtractor(RecordExtractor):
+    field_path: List[Union[InterpolatedString, str]]
+    config: Config
+    parameters: InitVar[Mapping[str, Any]]
+    decoder: Decoder = JsonDecoder(parameters={})
+
+    def __post_init__(self, parameters: Mapping[str, Any]):
+        for path_index in range(len(self.field_path)):
+            if isinstance(self.field_path[path_index], str):
+                self.field_path[path_index] = InterpolatedString.create(self.field_path[path_index], parameters=parameters)
+
+    def extract_records(self, response: requests.Response) -> List[Mapping[str, Any]]:
```

Comment:
This looks suspicious. Sure, I get it, it works, but I wonder if we can just replace that with a format matcher instead in the low-code declaration?

Reply:
I.e. I know that people run Xero connector in Cloud Builder — and they deal with dates somehow. They don't have access to custom components whatsoever. We could grab an example I guess 👀

Reply:
But it destructs the backward compatibility of those fields, instead of DateTime it would provide `"/Date(1630412754013+0000)/"`

Also, the incremental sync is built with one of these as cursor field
- "UpdatedDateUTC": "/Date(1630412754013+0000)/",

we need to convert each of those fields to standard datetime format,
Else on second thought, If incremental is removed, we only need the connector builder and no external strings needed.

Reply:
The response is just giving a timestamp with Date() wrapped around, 
- "UpdatedDateUTC": "/Date(1630412754013+0000)/",

CustomExtractor searches every nestings of response and converts it to standard date-time
If we remove it, It will change the schema and expected records and probably destination properties for anyone using it,

I've updated the Auth scheme to bearer, And every test is passing locally,
Is there any other way other than a custom component? It has to search every nested loop
Please comment if there are any possibilities.